src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/trust/TrustCorporateDto.java

```diff
@@ -267,19 +300,26 @@ public void setSaAddressRegion(String saAddressRegion) {
     }
 
     public AddressDto getServiceAddress() {
-        var serviceAddress = new AddressDto();
-        serviceAddress.setPropertyNameNumber(saAddressPremises);
-        serviceAddress.setLine1(saAddressLine1);
-        serviceAddress.setLine2(saAddressLine2);
-        serviceAddress.setCounty(saAddressRegion);
-        serviceAddress.setLocality(saAddressLocality);
-        serviceAddress.setCountry(saAddressCountry);
-        serviceAddress.setCareOf(saAddressCareOf);
-        serviceAddress.setPoBox(saAddressPoBox);
-        serviceAddress.setPostcode(saAddressPostalCode);
+        // If the address is being set from the sa fields generate the address
+        if (saAddressPostalCode != null) {
```

Comment:
Please see comment above, but would also like to double-check this is correct - could principal or service address details be present without postcodes?